### PR TITLE
Add recording of non VOI system vars to driver recording

### PR DESCRIPTION
**Pull request by hschilling**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 branches:
   only:
   - master
+  - add_system_vars_to_driver_recording
```
Comment (kmarsteller) [inline .travis.yml]: This should not remain in our real .travis.yml

File: openmdao/recorders/base_recorder.py [hunk 2]
```diff
@@ -247,11 +257,38 @@ def startup(self, object_requesting_recording):
...
+                # If MPI, and on rank 0, need to gather up all the variables
+                #    even those not local to rank 0
+                if MPI:
+                    all_vars = root.comm.gather(mysystem_outputs, root=0)
+                    if MPI.COMM_WORLD.rank == 0:
+                        mysystem_outputs = all_vars[-1]
+                        for d in all_vars[:-1]:
+                            mysystem_outputs.update(d)
+                # Filter out only the ones we want
+                # mysystem_outputs = {n for n in mysystem_outputs
```
Comment (kmarsteller) [inline openmdao/recorders/base_recorder.py]: Did we want the filtering commented out, or is this done elsewhere now?

File: .travis.yml (hunk 1 above)
Comment (hschilling) [reply .travis.yml]: fixed and pushed up again

File: openmdao/recorders/base_recorder.py (hunk 2 above)
Comment (hschilling) [reply openmdao/recorders/base_recorder.py]: fixed and pushed up

**Pull request merged**